### ISSUE TITLE
Help pane not rendering correctly when source=editor, but fine w/t source=console

### ISSUE THREAD

**Issue by skjerns** [spyder-ide/spyder#17709]
## Problem Description

When invoking the help pane from the editor, sometimes the docstrings are not rendered correctly. When inspecting the same object from the spyder/ipython console it works fine. This problem persists randomly with libraries using odd formatting.

### What steps reproduce the problem?

1. Install `mne` via `pip install mne`
2. Inspect e.g. `mne.stats.spatio_temporal_cluster_test` in editor
      ```
      import mne
      mne.stats.spatio_temporal_cluster_test
      ```
4. Resulting output is strangely formatted
5. Same object renders correctly when invoked from `console`

Problem is also present with other libraries, but I cannot remember right now which ones.
![image](https://user-images.githubusercontent.com/14980558/164198741-45798e24-c53b-4c03-8dd0-6749bf8ef102.png)

### What is the expected output? What do you see instead?

![image](https://user-images.githubusercontent.com/14980558/164198794-c7b9797c-d921-41b5-a813-9fa711d42901.png)

## Versions
* Spyder version: 5.3.0 None
* Python version: 3.9.7 64-bit
* Qt version: 5.12.10
* PyQt5 version: 5.12.3
* Operating System: Windows 10
* 
### Dependencies

```
# Mandatory:
atomicwrites >=1.2.0                          :  1.4.0 (OK)
chardet >=2.0.0                               :  4.0.0 (OK)
cloudpickle >=0.5.0                           :  2.0.0 (OK)
cookiecutter >=1.6.0                          :  1.7.3 (OK)
diff_match_patch >=20181111                   :  20200713 (OK)
intervaltree >=3.0.2                          :  3.1.0 (OK)
IPython >=7.31.1;<8.0.0                       :  7.32.0 (OK)
jedi >=0.17.2;<0.19.0                         :  0.18.1 (OK)
jellyfish >=0.7                               :  0.9.0 (OK)
jsonschema >=3.2.0                            :  4.4.0 (OK)
keyring >=17.0.0                              :  23.5.0 (OK)
nbconvert >=4.0                               :  6.4.4 (OK)
numpydoc >=0.6.0                              :  1.2 (OK)
paramiko >=2.4.0                              :  2.10.3 (OK)
parso >=0.7.0;<0.9.0                          :  0.8.3 (OK)
pexpect >=4.4.0                               :  4.8.0 (OK)
pickleshare >=0.4                             :  0.7.5 (OK)
psutil >=5.3                                  :  5.9.0 (OK)
pygments >=2.0                                :  2.11.2 (OK)
pylint >=2.5.0                                :  2.13.0 (OK)
pyls_spyder >=0.4.0                           :  0.4.0 (OK)
pylsp >=1.4.1;<1.5.0                          :  1.4.1 (OK)
pylsp_black >=1.2.0                           :  1.2.0 (OK)
qdarkstyle >=3.0.2;<3.1.0                     :  3.0.2 (OK)
qstylizer >=0.1.10                            :  0.2.1 (OK)
qtawesome >=1.0.2                             :  1.1.1 (OK)
qtconsole >=5.3.0;<5.4.0                      :  5.3.0 (OK)
qtpy >=2.0.1                                  :  2.0.1 (OK)
rtree >=0.9.7                                 :  0.9.7 (OK)
setuptools >=49.6.0                           :  58.0.4 (OK)
sphinx >=0.6.6                                :  4.4.0 (OK)
spyder_kernels >=2.3.0;<2.4.0                 :  2.3.0 (OK)
textdistance >=4.2.0                          :  4.2.2 (OK)
three_merge >=0.1.1                           :  0.1.1 (OK)
watchdog >=0.10.3                             :  2.1.7 (OK)
zmq >=17                                      :  22.3.0 (OK)

# Optional:
cython >=0.21                                 :  None (NOK)
matplotlib >=3.0.0                            :  3.5.1 (OK)
numpy >=1.7                                   :  1.22.3 (OK)
pandas >=1.1.1                                :  1.3.5 (OK)
scipy >=0.17.0                                :  1.7.3 (OK)
sympy >=0.7.3                                 :  None (NOK)

# Spyder plugins:
spyder_line_profiler.spyder.plugin 0.3.0.dev0 :  0.3.0.dev0 (OK)

```

**Comment by dalthviz** [comment]
Hi @skjerns thanks for the feedback! I'm was able to reproduce this but as you said seems like this kind of docs need the automatic import option (which is only available for the Console). Pinging @ccordoba12 just in case (maybe he can give more ideas about this).

**Comment by ccordoba12** [comment]
If the docstring for `spatio_temporal_cluster_test` is auto-generated at runtime, i.e. when the object is imported, then I'm afraid we won't be able to fix this because the libraries we use in the editor for this don't import code.

**Comment by rhkarls** [comment]
Just curious, is this the same reason the help pane cannot get documentation for some methods via the editor? 

Example below:

Cannot get `df.mean()` method documentation from editor, but works from console. `pd.DataFrame()` or `df.rename()` documentation does work from the editor however. 

![help_pane_editor](https://user-images.githubusercontent.com/2330659/164390689-bdc1fedd-9f2e-4d14-bab3-2b75f17a77e7.gif)

**Comment by skjerns** [comment]
Ah good catch! That actually happens quite often.

I think it would be great to have the console help for each object, however, I do understand if this is tricky to implement. Maybe a workaround could be to display the `console` help if available for a given object, and else the editor help?

Generally I found the editor help to be quite sluggish and buggy. It often does not find the docs or react at all, in which case I have to copy&paste the given var to the console and find help there. No I also understand why the console help is faster than the editor help.

**Comment by ccordoba12** [comment]
> Just curious, is this the same reason the help pane cannot get documentation for some methods via the editor?

Yep, I think so. Matplotlib also has the same problem, by the way.

> Maybe a workaround could be to display the console help if available for a given object, and else the editor help?

That's a great point! Technically it's not that easy (we need to develop a new completion provider connected to the current console), but it's feasible. We'll see if we can do it for Spyder 6.

**Comment by gepcel** [comment]
> I think it would be great to have the console help for each object, however, I do understand if this is tricky to implement. Maybe a workaround could be to display the console help if available for a given object, and else the editor help?

Until this is done in spyder, [this plugin](https://github.com/gepcel/ihelper) might be useful.

**Comment by skjerns** [comment]
> > I think it would be great to have the console help for each object, however, I do understand if this is tricky to implement. Maybe a workaround could be to display the console help if available for a given object, and else the editor help?
> 
> Until this is done in spyder, [this plugin](https://github.com/gepcel/ihelper) might be useful.

oh nice! that's exactly what I'm looking for
